airflow/api/common/experimental/mark_tasks.py

```diff
@@ -181,7 +181,39 @@ def set_state(task, execution_date, upstream=False, downstream=False,
         if len(sub_dag_ids) > 0:
             tis_altered += qry_sub_dag.all()
 
+    session.expunge_all()
     session.close()
 
     return tis_altered
 
+def set_dag_run_state(dag, execution_date, state=State.SUCCESS, commit=False):
```

Comment:
@saguziel There is a function called `set_dag_run_state` in `models.py`, but the logics are quite different. `set_dag_run_state` here basically called `set_state` above which is designed for mark success functionality and considers a lot more edge cases. But definitely the API functions need refactoring in the future to be consistent with other parts of the codebase.